File: src/server/src/main/java/com/project/application/models/post/PostViewResource.java
package com.project.application.models.post;

import com.project.domain.post.Post;
import io.micronaut.serde.annotation.Serdeable;

import java.time.ZonedDateTime;
import java.util.UUID;

@Serdeable
public record PostViewResource(UUID id, String imageUrl, ZonedDateTime createdAt, String createdBy,
                               String creatorImageUrl, String title, String text, String previewText, String categoryName,
                               String categoryNameTranslation, UUID categoryId) {
    public static PostViewResource from(Post post) {
        return new PostViewResource(post.getId(),
                post.getPostImages().get(0).getUrl(),
                post.getCreatedAt(),
                post.getUser().getFullName(), post.getUser().getImageUrl(),
                post.getTitle(), post.getText(), post.getPreviewText(), post.getCategory().getName(),
                post.getCategory().getTranslatedName(), post.getCategory().getId());
    }
}

File: src/server/src/main/java/com/project/application/models/post/PostViewResource.kt
package com.project.application.models.post

import com.project.domain.post.Post
import io.micronaut.serde.annotation.Serdeable
import java.time.ZonedDateTime
import java.util.UUID

@Serdeable
data class PostViewResource(val id: UUID, val imageUrl: String?, val createdAt: ZonedDateTime, val createdBy: String,
                       val creatorImageUrl: String?, val title: String?, val text: String?, val previewText: String?, val categoryName: String?,
                       val categoryNameTranslation: String?, val categoryId: UUID) {
    companion object {
        fun from(post: Post): PostViewResource {
            return PostViewResource(post.id,
                    post.postImages[0].url,
                    post.createdAt,
                    post.user!!.getFullName(), post.user!!.imageUrl,
                    post.title, post.text, post.previewText, post.category!!.name,
                    post.category!!.translatedName, post.category!!.id)
        }
    }
}
